base/linalg/dense.jl

```diff
@@ -74,10 +71,8 @@ julia> isposdef(A)
 true
 ```
 """
-function isposdef(A::AbstractMatrix{T}) where T
-    S = typeof(sqrt(one(T)))
-    isposdef!(S == T ? copy(A) : convert(AbstractMatrix{S}, A))
-end
+isposdef(A::AbstractMatrix{T}) where {T} =
+    isposdef!(copy_oftype(A, promote_type(typeof(chol(one(T))),Float32)))
```

Comment:
I'm a bit rusty on the promotion logic here but there is a difference between pivoted and non-pivoted since we don't yet have a generic version for the latter.